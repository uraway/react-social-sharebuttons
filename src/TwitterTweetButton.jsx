import React from 'react';

export default class TwitterTweetButton extends React.Component {
  constructor(props) {
    super(props);
    this.state = ({ initialized: false });
  }

  componentDidMount() {
    if (this.state.initialized) {
      return;
    }

    if (typeof twttr === 'undefined') {
      const twitterbutton = this.node;
      const twitterscript = document.createElement('script');
      twitterscript.src = '//platform.twitter.com/widgets.js';
      twitterscript.async = true;
      twitterscript.id = 'twitter-wjs';
      twitterbutton.parentNode.appendChild(twitterscript);
    } else {
      twttr.widgets.load(); // eslint-disable-line
    }

    this.initialized();
  }

  initialized() {
    this.setState({ initialized: true });
  }

  render() {
    return (
      <a
        ref={node => this.node = node}
        href="https://twitter.com/share"
        className="twitter-share-button"
        data-url={this.props.url}
        data-text={this.props.text}
        data-via={this.props.user}
        data-hashtags={this.props.hashtags}
      >
        Tweet
      </a>
    );
  }
}

TwitterTweetButton.propTypes = {
  url: React.PropTypes.string,
  text: React.PropTypes.string,
  hashtags: React.PropTypes.string,
  user: React.PropTypes.string
};

TwitterTweetButton.defaultProps = {
  url: (typeof window !== 'undefined' ? window.url : ''),
  text: '',
  hashtags: '',
  user: ''
};
/*
<a href="https://twitter.com/share" class="twitter-share-button" data-url="pageurl" data-text="pagetitle" data-via="uraway_" data-size="large" data-hashtags="hashtag" data-dnt="true">Tweet</a>
<script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+'://platform.twitter.com/widgets.js';fjs.parentNode.insertBefore(js,fjs);}}(document, 'script', 'twitter-wjs');</script>
*/
