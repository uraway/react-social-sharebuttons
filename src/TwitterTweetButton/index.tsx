import { FC, useEffect } from 'react';

declare let twttr: {
  widgets: {
    load: () => void;
  };
};

export type TwitterTweetButtonProps = {
  url?: string;
  text?: string;
  hashtags?: string;
  user?: string;
};

const TwitterTweetButton: FC<TwitterTweetButtonProps> = (props) => {
  useEffect(() => {
    let script: HTMLScriptElement;

    if (typeof twttr === 'undefined') {
      script = document.createElement('script');
      script.src = '//platform.twitter.com/widgets.js';
      script.async = true;
      script.id = 'twitter-wjs';
      document.getElementsByTagName('head')[0]?.appendChild(script);
    } else {
      twttr.widgets.load();
    }
  }, []);

  const { url, text, user, hashtags } = props;
  return (
    <a
      href="https://twitter.com/share"
      className="twitter-share-button"
      data-url={url}
      data-text={text}
      data-via={user}
      data-hashtags={hashtags}
    >
      Tweet
    </a>
  );
};

TwitterTweetButton.defaultProps = {
  url: typeof window !== 'undefined' ? window.location.href : '',
  text: '',
  hashtags: '',
  user: '',
};

export default TwitterTweetButton;
