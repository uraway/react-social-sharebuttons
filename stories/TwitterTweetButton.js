import React from 'react';
import { text } from '@storybook/addon-knobs';
import { TwitterTweetButton } from '../dist/index.esm.js';

export default {
  title: 'TwitterTweetButton',
};

export const Basic = () => (
  <TwitterTweetButton
    url={text('url', '')}
    text={text('text', '')}
    hashtags={text('hashtags', 'sharebutton')}
    user={text('user', 'uraway_')}
  />
);
