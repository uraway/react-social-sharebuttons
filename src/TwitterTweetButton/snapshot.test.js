import React from 'react';
import renderer from 'react-test-renderer';
import TwitterTweetButton from '.';

it('renders correctly with required props', () => {
  const tree = renderer.create(<TwitterTweetButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with other props', () => {
  const tree = renderer
    .create(<TwitterTweetButton url="url" hashtags={['hashtag']} text="text" user="user" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
