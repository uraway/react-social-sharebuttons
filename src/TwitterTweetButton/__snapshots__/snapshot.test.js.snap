// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with other props 1`] = `
<div>
  <a
    className="twitter-share-button"
    data-hashtags={
      Array [
        "hashtag",
      ]
    }
    data-text="text"
    data-url="url"
    data-via="user"
    href="https://twitter.com/share"
  >
    Tweet
  </a>
</div>
`;

exports[`renders correctly with required props 1`] = `
<div>
  <a
    className="twitter-share-button"
    data-hashtags=""
    data-text=""
    data-via=""
    href="https://twitter.com/share"
  >
    Tweet
  </a>
</div>
`;
